enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/task/cleaning/ActivationCleaningService.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2022 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.onboardingserver.task.cleaning;
+
+import com.wultra.app.onboardingserver.common.database.OnboardingProcessRepository;
+import com.wultra.app.onboardingserver.common.database.entity.OnboardingProcessEntity;
+import com.wultra.app.onboardingserver.common.errorhandling.RemoteCommunicationException;
+import com.wultra.app.onboardingserver.impl.service.ActivationService;
+import com.wultra.security.powerauth.client.v3.ActivationStatus;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Service to cleaning activations.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Service
+@Slf4j
+class ActivationCleaningService {
+
+    private static final int BATCH_SIZE = 100;
+
+    private final OnboardingProcessRepository onboardingProcessRepository;
+
+    private final ActivationService activationService;
+
+    @Autowired
+    public ActivationCleaningService(
+            final OnboardingProcessRepository onboardingProcessRepository,
+            final ActivationService activationService) {
+
+        this.onboardingProcessRepository = onboardingProcessRepository;
+        this.activationService = activationService;
+    }
+
+    /**
+     * Cleanup activations of failed onboarding processes.
+     */
+    public void cleanupActivations() {
```

Comment:
Missing `@Transactional` on purpose. It is possible to load processes anyway. We do not want to do REST calls in transaction. Each process is updated in a separate transaction, because `save` in Spring Data is annotated by default.